src/assets/sass/protocol/components/_footer.scss

```diff
@@ -12,111 +12,231 @@
+}
 
-    li {
-        list-style-type: none;
-        margin: 0 0 $margin-md;
+/* -------------------------------------------------------------------------- */
+// Footer Mozilla logo
+
+.mzp-c-footer-primary-logo {
+    margin-bottom: $spacing-xl;
+
+    a {
+        background: url('#{$image-path}/logos/mozilla/white.svg') no-repeat;
+        @include background-size(100px, 32px);
+        @include image-replaced;
+        display: inline-block;
+        height: 32px;
+        width: 100px;
+    }
+}
+
+
+/* -------------------------------------------------------------------------- */
+// Footer section containing list and heading
+
+.mzp-c-footer-sections {
+    @include clearfix;
+
+    @media #{$mq-md} {
+        border-bottom: 1px solid $color-gray-70;
+        padding-bottom: $layout-lg;
+    }
+
+    @media #{$mq-lg} {
+        padding-bottom: $layout-md;
+    }
+}
+
+.mzp-c-footer-section {
+    @include clearfix;
+    margin-bottom: $spacing-sm;
+
+    a:link,
+    a:visited {
+        text-decoration: none;
     }
 
-    nav {
-        @include border-box;
-        @include clearfix;
-        margin: 0 auto;
-        min-width: $content-xs;
+    a:hover,
+    a:focus,
+    a:active {
+        text-decoration: underline;
+    }
+
+    @media (min-width: 480px) and (max-width: 1055px) {
+        @include bidi(((float, left, right),));
+        @include grid-half;
+
+        &:nth-child(odd) {
+            @include bidi(((clear, left, right),));
+            @include bidi(((padding, 0 ($layout-md / 2) 0 0, 0 0 0 ($layout-md / 2)),));
+        }
+
+        &:nth-child(even) {
+            @include bidi(((padding, 0 0 0 ($layout-md / 2), 0 ($layout-md / 2) 0 0),));
+        }
     }
+}
+
+@media #{$mq-lg} {
 
     .mzp-c-footer-section {
-        margin-bottom: $margin-xl;
+        @include bidi(((float, left, right),));
+        padding: 0 ($layout-md / 2);
+
+        &:first-child {
+            @include bidi(((padding, 0 ($layout-md / 2) 0 0, 0 0 0 ($layout-md / 2)),));
+        }
+
+        &:last-child {
+            @include bidi(((padding, 0 0 0 ($layout-md / 2), 0 ($layout-md / 2) 0 0),));
+        }
+    }
+
+    @for $cols from 2 through 6 {
+        $width: 100% / $cols;
+        // http://lea.verou.me/2011/01/styling-children-based-on-their-number-with-css3/
+        .mzp-c-footer-section:first-child:nth-last-child(n+#{$cols}),
+        .mzp-c-footer-section:first-child:nth-last-child(n+#{$cols}) ~ .mzp-c-footer-section:nth-child(1n-1) {
+            width: calc(#{$width} - (#{$layout-md} - (#{$layout-md} / #{$cols})));
+        }
     }
+}
 
-    .mzp-c-footer-primary {
+.mzp-c-footer-heading {
+    &,
+    & button {
         @include text-body-md;
+        background-color: transparent;
+        border: 0;
+        border-bottom: 1px solid $color-gray-70;
+        color: inherit;
+        margin-bottom: 0;
+        padding: $spacing-md 0;
+        position: relative;
+        @include bidi(((text-align, left, right),));
```

Comment:
`text-align` usually goes between `position` and `width` in the dictionary 😜 Do you want the includes all grouped at the top?

Reply:
Yes, I'm just suggesting we follow convention for how we do this elsewhere. I'd rather give up ordering decisions to my text editor, rather than us have to start to maintain these kind of things manually 😜